build_runner/lib/src/generate/watch_impl.dart

```diff
@@ -157,4 +157,4 @@ class WatchImpl {
 
 Map<AssetId, ChangeType> _collectChanges(List<List<AssetChange>> changes) =>
     new Map.fromIterable(changes.expand((c) => c),
-        key: (c) => c.id, value: (c) => c.type);
+        key: (AssetChange c) => c.id, value: (AssetChange c) => c.type);
```

Comment:
ack, this should be inferred as well if the iterable had a generic type in `Map.fromIterable`.

I think its worth noting all these and filing a bug on the core libs to add generic types.